### ISSUE TITLE
Device hangs when retrieving many places

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/prox#543]
I think this may be related to waiting for `Promise.allFilled` for travel times.

**Comment by mcomella** [comment]
I am unable to reproduce now (both on my device and in the simulator). Thoughts:
* Maria & antlam seem to be able to reproduce but Anthony can't – perhaps this is a device issue?
* To test if it's travel times, we can [short-circuit here](https://github.com/mozilla-mobile/prox/blob/e24a8858d09f558af4e19770770efe428f5725e6/Prox/Prox/Data/TravelTimesProvider.swift#L62).
  * It's not actually Promise.allFilled, but we wait for all providers to return before beginning our next request
* Since we can't reproduce this week, Apple's API may have been less functional last week – perhaps we should add a preventative timeout to prevent errors in the future (it is pretty crappy to block loading on travel times).

**Comment by mcomella** [comment]
Repro notes:
* Maria mentioned this typically occurs for ~1hr after she has received a new BB
  * I wonder if Apple has special perms for new apps and routing
  * I wonder if we make a lot of routing requests when new BB are released so Apple is limiting our bundle ID
* Maria, Gemma & I repro'd at the same time. When I no longer could repro, Maria could not repro (didn't check with Gemma)
* I can only repro on device (I tried to repro with the Network Link Conditioner & Simulator but I couldn't)

Related?
* I repro'd once on device. Looking at the logs, Firebase loaded correctly (e.g. "dropping place...") but I never saw the logs from MapKit about rate limiting. The last log, after Firebase dropped places, was:

> 2017-03-03 15:50:48.342973 Prox Mock Location[442:27571] [LogMessageLogging] 74.1 Unable to retrieve CarrierName. CTError: domain-2, code-5, errStr:((os/kern) failure)

Unrelated?
* The logs show unsatisfiable constraints but I think that may be when the view is shown

**Comment by mcomella** [comment]
via [`MKDirections.calculateETA`]():
> Your completion handler is executed on your app’s main thread.

I wonder if we're hitting a race! Maybe we all hit it at the same time because the server is taking longer to respond than usual.

**Comment by mcomella** [comment]
I saw this occur on Maria's iPhone 6 but not in my iPhone 7+ in the same moment – maybe this is affected by CPU speed (which is why we haven't been able to reproduce frequently on laptops).

**Comment by mcomella** [comment]
A few more repro notes from Maria: can repro more frequently on cell network. Happens more earlier in the day

**Comment by liuche** [comment]
Saw this once in the simulator, but the logs don't seem to be more helpful than showing the error for geo/direction throttle.

<img width="1069" alt="screenshot 2017-03-07 13 51 29" src="https://cloud.githubusercontent.com/assets/1082253/23681201/e38de0b6-0342-11e7-980c-7cf808dcf3ae.png">

**Comment by thebnich** [comment]
I found that `MKDirections.calculateETA` simply isn't executing its callback at all, which seems like an iOS bug. I added a delayed dispatch on the main thread to see if maybe the main thread is deadlocked and we're seeing this as a side effect, but the dispatched callback executes just fine after the app freezes.

Was originally going to sort by raw distance and update the walking distance only as the cards are shown, but that would slightly reorder the cards, which is risky given the closeness to Chicago. Adding a timeout is probably the safest bet for now since it doesn't change the experience.
